c7n/resources/shield.py

```diff
@@ -57,8 +57,15 @@ def get_protections_paginator(client):
 
 
 def get_type_protections(client, model):
-    protections = get_protections_paginator(
-        client).paginate().build_full_result().get('Protections')
+    try:
+        protections = get_protections_paginator(
+            client).paginate().build_full_result().get('Protections')
+    except ClientError as e:
```

Comment:
```
except client.exceptions.ResourceNotFoundException:
      return []
```

we should try to adopt modern boto3 exception handling for new prs, its smaller and more explicit.